pkg/api/types.go

```diff
@@ -221,6 +224,24 @@ type GCEPersistentDiskVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// A ISCSI Disk can only be mounted as read/write once.
+type ISCSIDiskVolumeSource struct {
+	// Required: iSCSI target IP
+	TargetIP string `json:"targetIP,omitempty"`
+	// Required:  target iSCSI Qualified Name
+	IQN string `json:"iqn,omitempty"`
+	// Required: iSCSI target lun number
+	Lun int `json:"lun,omitempty"`
+	// Required: Filesystem type to mount.
```

Comment:
auto-detect works for some (e.g. extX) but not all.  xfs, etc needs explicit specification.